nucliadb/nucliadb/common/maindb/tikv.py

```diff
@@ -262,21 +264,33 @@ async def count(self, match: str) -> int:
 class TiKVTransaction(Transaction):
     driver: TiKVDriver
 
-    def __init__(self, txn: Any, driver: TiKVDriver):
+    def __init__(
+        self,
+        txn: Any,
+        driver: TiKVDriver,
+        replication_tx: Optional[PGTransaction] = None,
+    ):
         self.txn = txn
         self.driver = driver
         self.data_layer = TiKVDataLayer(txn)
         self.open = True
+        self.replication_tx = replication_tx
+        self.replication_ops: List[int] = []
 
     async def abort(self):
         if not self.open:
             return
         await self.data_layer.abort()
+        if self.replication_tx:
+            await self.replication_tx.abort()
         self.open = False
 
     async def commit(self):
         assert self.open
         await self.data_layer.commit()
+        if self.replication_tx:
+            await self.replication_tx.create_replication_commit(self.replication_ops)
```

Comment:
I would check if self.replication_ops has some element, just in case?

Reply:
Adding an empty array is safe, but a waste. I'll add the check.